### PR TITLE
MHV-43607 VA.gov SM: Update the message threads with the new accordion designs

### PR DESCRIPTION
**Pull request by alex-morgun-1**
## Summary

VA.gov SM: Update the message threads with the new accordion designs

User Story: As an SM user, I want to be able to view an entire thread, so that I can see all my messages in the thread.

Acceptance Criteria:
AC1 New accordion message thread design has been implemented
AC2 Support the expand all button including all the message read api calls

## Related issue(s)
- [VA Jira MHV-43607 VA.gov SM: Update the message threads with the new accordion designs](https://vajira.max.gov/browse/MHV-43607)

## Testing done

- Unit tests
- Cypress Tests
- SQA Testing

## Screenshots
_Note: This field is mandatory for component work and UI changes (non-component work should NOT have screenshots)._

| | Before | After |
| --- | --- | --- |
| Mobile | image | image |
| Desktop |image | image |

## What areas of the site does it impact?
My Health - Secure Messaging / Thread View

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

File: src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadItem.jsx [hunk 1]
```diff
@@ -23,104 +40,82 @@ const MessageThreadItem = props => {
...
+  //     <>
+  //       <i
+  //         className="fas fa-paperclip"
+  //         aria-hidden
+  //         style={{ visibility: attachments !== true ? 'hidden' : '' }}
+  //       />
+  //       {from}
+  //     </>
+  //   );
+  // };
```
Comment (vsaleem) [inline src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadItem.jsx:74]: Do we need to keep this?

File: src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadItem.jsx [hunk 2]
```diff
@@ -23,104 +40,82 @@ const MessageThreadItem = props => {
...
+  //   () => {
+  //     return `${!isRead ? 'New' : ''} message ${
+  //       attachment ? 'with attachment' : ''
+  //     } from ${senderName}, ${dateFormat(
+  //       sentDate,
+  //       'MMMM D, YYYY [at] h:mm a z',
+  //     )}. Click to ${isExpanded ? 'Collapse message' : 'Expand message'}`;
+  //   },
+  //   [attachment, isExpanded, isRead, senderName, sentDate],
+  // );
```
Comment (vsaleem) [inline src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadItem.jsx:86]: Same for this one.. maybe put this in a helper file if we need this later.

File: src/applications/mhv/secure-messaging/tests/components/MessageActionButtons/PrintBtn.unit.spec.jsx [hunk 3]
```diff
@@ -11,8 +11,16 @@ describe('Print button', () => {
   beforeEach(() => {
     const id = 7155731;
     const handlePrint = () => {};
+    // const initialState = {
+    //   message: { message: { messageResponse }, messages: null },
+    // };
```
Comment (vsaleem) [inline src/applications/mhv/secure-messaging/tests/components/MessageActionButtons/PrintBtn.unit.spec.jsx:16]: Same

File: src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadItem.jsx (hunk 1 above)
Comment (oleksii-morgun-va) [reply src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadItem.jsx:74]: yes, this should stay until accordion is ready to accept nodes in the header and subheader

File: src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadItem.jsx (hunk 2 above)
Comment (oleksii-morgun-va) [reply src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadItem.jsx:86]: this is pending confirmation from UCD on what aria labels should be reading like once the accordion accepts icons.

**Review by vsaleem — APPROVED** [review]
It looks like a cypress test might be failing; please check on your end.. if everything passes then you're good to go!

**Pull request merged**